lib/CodeGen/CodeGenFunction.h

```diff
@@ -1238,6 +1238,48 @@ class CodeGenFunction : public CodeGenTypeCache {
   const TargetInfo &getTarget() const { return Target; }
   llvm::LLVMContext &getLLVMContext() { return CGM.getLLVMContext(); }
 
+public:
```

Comment:
There needs to be a comment in the code that describes the overall scheme that you are using here during code generation and why you choose to use it.